src/CachePeers.cc

```diff
@@ -82,11 +82,21 @@ AbsorbConfigured(std::unique_ptr<CachePeer> &&peer)
     peerClearRRStart();
 }
 
+void
+DeleteConfigured(Configuration::SmoothReconfiguration &sr, CachePeer * const peer)
+{
+    peerSelectDrop(sr, *peer);
+    Assure(Config.peers);
+    Config.peers->remove(peer);
+}
+
 void
 DeleteConfigured(CachePeer * const peer)
 {
     Assure(Config.peers);
     Config.peers->remove(peer);
+    // XXX: peer is deleted by now, so pass it as shared pointer
+    peerSelectDrop(*peer);
```

Comment:
I have found another case where this problematic code is triggered -- async neighbors_init() call that was scheduled at the end of _smooth_ reconfiguration after `Component<CachePeers*>::Reconfigure()` adds a new cache_peer. I adjusted PR description accordingly. I am still working on a plan to address this XXX.